examples/fluids/src/setupts.c

```diff
@@ -164,6 +166,7 @@ static PetscErrorCode Surface_Forces_NS(DM dm, Vec G_loc, PetscInt num_walls, co
         const PetscInt           p = points[i];
         const StateConservative *r;
         PetscCall(DMPlexPointLocalRead(dm, p, g, &r));
+        PetscCall(PetscSectionGetDof(s, p, &dof));
```

Comment:
Now you know how many dofs, but still need to loop over the nodes.

Reply:
Once you're pretty sure it's working, can you run a coarse mesh with cubic elements to produce a new lift and drag plot?

Reply:
Yes gladly! I will produce a new lift/drag plot. Thanks.

Reply:
Great. You have to fix this bit first (or the results will be quite wrong). Each node has 5 dofs, so loop over the nodes at each point and read out their momentum residuals. The code here reads only the first node, which is correct at vertices, but is only part of a (cubic) edge or face.